src/Symfony/Component/Mailer/Transport/NativeTransportFactory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Transport;
+
+use Symfony\Component\Mailer\Exception\TransportException;
+use Symfony\Component\Mailer\Exception\UnsupportedSchemeException;
+use Symfony\Component\Mailer\Transport\Smtp\SmtpTransport;
+use Symfony\Component\Mailer\Transport\Smtp\Stream\SocketStream;
+
+/**
+ * @author Laurent VOULLEMIER <laurent.voullemier@gmail.com>
+ */
+final class NativeTransportFactory extends AbstractTransportFactory
+{
+    public function create(Dsn $dsn): TransportInterface
+    {
+        if (!\in_array($dsn->getScheme(), $this->getSupportedSchemes(), true)) {
+            throw new UnsupportedSchemeException($dsn, 'native', $this->getSupportedSchemes());
+        }
+
+        $isWindows = 'WIN' === strtoupper(substr(PHP_OS, 0, 3));
+        if ($sendMailPath = ini_get('sendmail_path')) {
+            if (!$isWindows || 'native+sendmail' === $dsn->getScheme() || 'native' === $dsn->getScheme()) {
+                return new SendmailTransport($sendMailPath, $this->dispatcher, $this->logger);
+            }
+        }
+
+        if (!$isWindows || 'native+sendmail' === $dsn->getScheme()) {
+            throw new TransportException('sendmail_path is not configured in php.ini.');
+        }
+
+        // Only for windows hosts; at this point non-windows
+        // host have already thrown an exception or returned a transport
+        $host = ini_get('smtp');
+        $port = (int) ini_get('smtp_port');
+
+        if (!$host || !$port) {
+            throw new TransportException('smtp or smtp_port is not configured in php.ini.');
+        }
+
+        $socketStream = new SocketStream();
+        $socketStream->setHost($host);
+        $socketStream->setPort($port);
+        if (465 !== $port) {
+            $socketStream->disableTls();
+        }
+
+        return new SmtpTransport($socketStream, $this->dispatcher, $this->logger);
+    }
+
+    protected function getSupportedSchemes(): array
+    {
+        $supportedShemes = ['native', 'native+sendmail'];
```

Comment:
So we really need the 2 possibiliies? Can't we only support `native` and do the switch ourselves in the code depending on the platform?

Reply:
Not really needed indeed; I added the two options to be consistent with `SendmailTransportFactory`. But I can only keep `native` if you prefer.

Reply:
I would remove the `native-sendmail` scheme, which is not needed IMHO.